appserver/security/core-ee/src/main/java/com/sun/enterprise/security/ee/authentication/glassfish/jdbc/JDBCRealm.java

```diff
@@ -425,8 +425,9 @@ private char[] hashPassword(char[] password) throws CharacterCodingException {
         if (HEX.equalsIgnoreCase(encoding)) {
             result = hexEncode(bytes);
         } else if (BASE64.equalsIgnoreCase(encoding)) {
-            result = base64Encode(bytes).toCharArray();
-        } else { // no encoding specified
+            result = Base64.getUrlEncoder().encodeToString(bytes).toCharArray();
```

Comment:
Try to use Basic Base64 aphabet (`getEncoder()` instead of `getUrlEncoder()`).

Or change `glassfish/appserver/tests/appserv-tests/devtests/security/jdbcrealm/util/PopulateDB.java`.

Reply:
Thanks, that should be that... I knew it was something like that but forgot details and was lazy to find the difference.
I have to fix it on other places too.